### PR TITLE
update: refactor logic store blocks

### PR DESCRIPTION
**Pull request by tu-mavan-ncc**
# Ticket
`<paste ticket link here>`

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: block/slot.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package block
+
+type SlotInfo struct {
+	Slot          uint64   `json:"slot"`
+	Hash          [32]byte `json:"hash"`
+	PrevHash      [32]byte `json:"prev_hash"`
+	LastEntryHash [32]byte `json:"last_entry_hash"` // hash of the last entry in the block, used for PoH
+	LeaderID      string   `json:"leader_id"`
```
Comment (hoangdohuy) [inline block/slot.go:7]: có cần cái này không nhỉ

File: block/slot.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package block
+
+type SlotInfo struct {
+	Slot          uint64   `json:"slot"`
+	Hash          [32]byte `json:"hash"`
+	PrevHash      [32]byte `json:"prev_hash"`
+	LastEntryHash [32]byte `json:"last_entry_hash"` // hash of the last entry in the block, used for PoH
```
Comment (hoangdohuy) [inline block/slot.go:6]: `LastEntryHash` vs `Hash`

File: validator/validator.go [hunk 3]
```diff
@@ -188,49 +188,46 @@ func (v *Validator) ReadyToStart(slot uint64) bool {
...
-			)
-
-			blk.Sign(v.PrivKey)
-			logx.Info("VALIDATOR", fmt.Sprintf("Leader assembled block: slot=%d, entries=%d", lastSlot, len(v.collectedEntries)))
-			prevBlock := v.blockStore.Block(prevSlot)
-			if prevBlock != nil {
-				monitoring.RecordBlockTime(blk.CreationTimestamp().Sub(prevBlock.CreationTimestamp()))
-			}
+		// Assemble block
+		prevSlot := lastSlot - 1
```
Comment (hoangdohuy) [inline validator/validator.go:203]: Chắc đoạn này cần check lại sao lại phải bỏ câu if ở đây.
Trước mình thêm để fix bug miss block lúc chạy nhiều node thì phải.

File: validator/validator.go [hunk 4]
```diff
@@ -188,49 +188,46 @@ func (v *Validator) ReadyToStart(slot uint64) bool {
 // handleEntry buffers entries and assembles a block at slot boundary.
 func (v *Validator) handleEntry(entries []poh.Entry) {
 	currentSlot := v.Recorder.CurrentSlot()
+	v.Mempool.SetCurrentSlot(currentSlot)
```
Comment (hoangdohuy) [inline validator/validator.go]: slot trong mempool phải là slot của toàn mạng, tức là phải được set ở đoạn đã đủ vote thôi, như hiện tại đã được set ở `applyDataToBlock`
